From fb96303006dad970cd062c34bc3848987cb8a219 Mon Sep 17 00:00:00 2001
From: Konrad Leszczynski <konrad.leszczynski@intel.com>
Date: Tue, 15 Dec 2015 12:19:35 +0000
Subject: [PATCH 11/12] usb: gadget: midi: add function to provide alsa config

Needed only to keep same user-space interface as original Google patch

Change-Id: I4059e4a097661395c556d1bbe6b8a3f06ed34c76
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-3355
Signed-off-by: Mian Yousaf Kaukab <yousaf.kaukab@intel.com>
Signed-off-by: Konrad Leszczynski <konrad.leszczynski@intel.com>
Reviewed-on: https://android.intel.com:443/448842
---
 drivers/usb/gadget/f_midi.c | 8 ++++++++
 drivers/usb/gadget/u_midi.h | 2 ++
 2 files changed, 10 insertions(+)

diff --git a/drivers/usb/gadget/f_midi.c b/drivers/usb/gadget/f_midi.c
index 4c7cb45..96629e6 100644
--- a/drivers/usb/gadget/f_midi.c
+++ b/drivers/usb/gadget/f_midi.c
@@ -94,6 +94,14 @@ static inline struct f_midi *func_to_midi(struct usb_function *f)
 	return container_of(f, struct f_midi, func);
 }
 
+void midi_get_alsa_config(struct usb_function *f, int *card, int *device)
+{
+	struct f_midi *midi = func_to_midi(f);
+
+	*card = midi->rmidi->card->number;
+	*device = midi->rmidi->device;
+}
+
 static void f_midi_transmit(struct f_midi *midi, struct usb_request *req);
 
 DECLARE_UAC_AC_HEADER_DESCRIPTOR(1);
diff --git a/drivers/usb/gadget/u_midi.h b/drivers/usb/gadget/u_midi.h
index 63f0522..dd38f09 100644
--- a/drivers/usb/gadget/u_midi.h
+++ b/drivers/usb/gadget/u_midi.h
@@ -36,4 +36,6 @@ struct f_midi_opts {
 	 int				refcnt;
 };
 
+void midi_get_alsa_config(struct usb_function *f, int *card, int *device);
+
 #endif /* U_MIDI_H */
-- 
1.9.1

